File: src/ros2_demo_py/ros2_demo_py/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

# create class to excute
class RobotNewsStationNode(Node):
    def __init__(self):
        self.counter_ = 0
        super().__init__("robot_news_station")

        self.robot_name_ = "C3P0"
        self.publisher_ = self.create_publisher(String, "News", 10)
        self.timer_ = self.create_timer(1, self.publisher_news)
        self.get_logger().info("This is robot news has been started")

    def publisher_news(self):
        msg = String()
        msg.data = "This is news for" + str(self.robot_name_) + " time(s) " + str(self.counter_)
        self.publisher_.publish(msg)
        self.counter_ += 1

def main(args = None):
    rclpy.init(args=args)
    node = RobotNewsStationNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/ros2_demo_cpp/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"
// #include "rclcpp/timer.hpp"

class RobotNewsStationNode: public rclcpp::Node
{
public:
    RobotNewsStationNode() : Node("robot_news_station"),
                             counter_(0),
                             robot_name_("R2D2"){
        publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
        timer_ = this->create_wall_timer(std::chrono::seconds(1),
                                        std::bind(&RobotNewsStationNode::publish_news, this));
        RCLCPP_INFO(this->get_logger(), "Robot Station has been started");
    }

private:
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;

    void publish_news(){
        auto msg = example_interfaces::msg::String();
        msg.data = "Robot publishes the name is "+ robot_name_ + " times " + std::to_string(counter_);
        publisher_->publish(msg);
        counter_++;
    }

    rclcpp::TimerBase::SharedPtr timer_;
    std::string robot_name_;
    int counter_;
};


int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>();
    
    rclcpp::spin(node);

    rclcpp::shutdown();

    return 0;
}